Just please fix this warning/error:
```
../librz/core/visual.c: In function ‘rz_core_visual_disasm_down’:
../librz/core/visual.c:4056:22: error: variable ‘f’ set but not used [-Werror=unused-but-set-variable]
  RzAnalysisFunction *f = NULL;
                      ^
```

--- a/librz/analysis/serialize_analysis.c
+++ b/librz/analysis/serialize_analysis.c
@@ -13,13 +13,13 @@
  *
  * /
  *   /blocks
- *     0x<addr>={size:<ut64>, jump?:<ut64>, fail?:<ut64>, traced?:true, folded?:true, colorize?:<ut32>,
+ *     0x<addr>={size:<ut64>, jump?:<ut64>, fail?:<ut64>, traced?:true, colorize?:<ut32>,
  *               fingerprint?:"<base64>", diff?: <RzAnalysisDiff>, switch_op?:<RzAnalysisSwitchOp>,
  *               ninstr:<int>, op_pos?:[<ut16>], stackptr:<int>, parent_stackptr:<int>,
  *               cmpval:<ut64>, cmpreg?:<str>}
  *   /functions
  *     0x<addr>={name:<str>, bits?:<int>, type:<int>, cc?:<str>, stack:<int>, maxstack:<int>,
- *               ninstr:<int>, folded?:<bool>, pure?:<bool>, bp_frame?:<bool>, bp_off?:<st64>, noreturn?:<bool>,
+ *               ninstr:<int>, pure?:<bool>, bp_frame?:<bool>, bp_off?:<st64>, noreturn?:<bool>,
  *               fingerprint?:"<base64>", diff?:<RzAnalysisDiff>, bbs:[<ut64>], imports?:[<str>], vars?:[<RzAnalysisVar>],
  *               labels?: {<str>:<ut64>}}
  *   /xrefs
@@ -254,9 +254,6 @@ static void block_store(RZ_NONNULL Sdb *db, const char *key, RzAnalysisBlock *bl
 	if (block->traced) {
 		pj_kb (j, "traced", true);
 	}
-	if (block->folded) {
-		pj_kb (j, "folded", true);
-	}
 	if (block->colorize) {
 		pj_kn (j, "colorize", (ut64)block->colorize);
 	}
@@ -329,7 +326,6 @@ enum {
 	BLOCK_FIELD_JUMP,
 	BLOCK_FIELD_FAIL,
 	BLOCK_FIELD_TRACED,
-	BLOCK_FIELD_FOLDED,
 	BLOCK_FIELD_COLORIZE,
 	BLOCK_FIELD_FINGERPRINT,
 	BLOCK_FIELD_DIFF,
@@ -392,12 +388,6 @@ static bool block_load_cb(void *user, const char *k, const char *v) {
 				break;
 			}
 			proto.traced = child->num.u_value;
-			break;
-		case BLOCK_FIELD_FOLDED:
-			if (child->type != RZ_JSON_BOOLEAN) {
-				break;
-			}
-			proto.folded = child->num.u_value;
 			break;
 		case BLOCK_FIELD_COLORIZE:
 			if (child->type != RZ_JSON_INTEGER) {
@@ -516,7 +506,6 @@ static bool block_load_cb(void *user, const char *k, const char *v) {
 	block->jump = proto.jump;
 	block->fail = proto.fail;
 	block->traced = proto.traced;
-	block->folded = proto.folded;
 	block->colorize = proto.colorize;
 	block->fingerprint = proto.fingerprint;
 	block->diff = proto.diff;
@@ -551,7 +540,6 @@ RZ_API bool rz_serialize_analysis_blocks_load(RZ_NONNULL Sdb *db, RZ_NONNULL RzA
 	key_parser_add (ctx.parser, "jump", BLOCK_FIELD_JUMP);
 	key_parser_add (ctx.parser, "fail", BLOCK_FIELD_FAIL);
 	key_parser_add (ctx.parser, "traced", BLOCK_FIELD_TRACED);
-	key_parser_add (ctx.parser, "folded", BLOCK_FIELD_FOLDED);
 	key_parser_add (ctx.parser, "colorize", BLOCK_FIELD_COLORIZE);
 	key_parser_add (ctx.parser, "fingerprint", BLOCK_FIELD_FINGERPRINT);
 	key_parser_add (ctx.parser, "diff", BLOCK_FIELD_DIFF);
@@ -886,9 +874,6 @@ static void function_store(RZ_NONNULL Sdb *db, const char *key, RzAnalysisFuncti
 	pj_ki (j, "stack", function->stack);
 	pj_ki (j, "maxstack", function->maxstack);
 	pj_ki (j, "ninstr", function->ninstr);
-	if (function->folded) {
-		pj_kb (j, "folded", true);
-	}
 	if (function->bp_frame) {
 		pj_kb (j, "bp_frame", true);
 	}
@@ -971,7 +956,6 @@ enum {
 	FUNCTION_FIELD_STACK,
 	FUNCTION_FIELD_MAXSTACK,
 	FUNCTION_FIELD_NINSTR,
-	FUNCTION_FIELD_FOLDED,
 	FUNCTION_FIELD_PURE,
 	FUNCTION_FIELD_BP_FRAME,
 	FUNCTION_FIELD_BP_OFF,
@@ -1055,12 +1039,6 @@ static bool function_load_cb(void *user, const char *k, const char *v) {
 				break;
 			}
 			function->ninstr = (int)child->num.s_value;
-			break;
-		case FUNCTION_FIELD_FOLDED:
-			if (child->type != RZ_JSON_BOOLEAN) {
-				break;
-			}
-			function->folded = child->num.u_value ? true : false;
 			break;
 		case FUNCTION_FIELD_PURE:
 			if (child->type != RZ_JSON_BOOLEAN) {
@@ -1231,7 +1209,6 @@ RZ_API bool rz_serialize_analysis_functions_load(RZ_NONNULL Sdb *db, RZ_NONNULL 
 	key_parser_add (ctx.parser, "stack", FUNCTION_FIELD_STACK);
 	key_parser_add (ctx.parser, "maxstack", FUNCTION_FIELD_MAXSTACK);
 	key_parser_add (ctx.parser, "ninstr", FUNCTION_FIELD_NINSTR);
-	key_parser_add (ctx.parser, "folded", FUNCTION_FIELD_FOLDED);
 	key_parser_add (ctx.parser, "pure", FUNCTION_FIELD_PURE);
 	key_parser_add (ctx.parser, "bp_frame", FUNCTION_FIELD_BP_FRAME);
 	key_parser_add (ctx.parser, "bp_off", FUNCTION_FIELD_BP_OFF);

--- a/librz/core/agraph.c
+++ b/librz/core/agraph.c
@@ -4044,21 +4044,6 @@ static void rotateColor(RzCore *core) {
 	rz_config_set_i (core->config, "scr.color", color);
 }
 
-// dupe in visual.c
-static bool toggle_bb(RzCore *core, ut64 addr) {
-	RzAnalysisFunction *fcn = rz_analysis_get_fcn_in (core->analysis, addr, RZ_ANALYSIS_FCN_TYPE_NULL);
-	if (fcn) {
-		RzAnalysisBlock *bb = rz_analysis_fcn_bbget_in (core->analysis, fcn, addr);
-		if (bb) {
-			bb->folded = !bb->folded;
-		} else {
-			rz_warn_if_reached ();
-		}
-		return true;
-	}
-	return false;
-}
-
 static char *get_graph_string(RzCore *core, RzAGraph *g) {
 	int c = rz_config_get_i (core->config, "scr.color");
 	int u = rz_config_get_i (core->config, "scr.utf8");
@@ -4360,18 +4345,6 @@ RZ_API int rz_core_visual_graph(RzCore *core, RzAGraph *g, RzAnalysisFunction *_
 		case 'V':
 			if (fcn) {
 				agraph_toggle_callgraph (g);
-			}
-			break;
-		case 'Z':
-			if (okey == 27) { // shift-tab
-				agraph_prev_node (g);
-			} else {
-				RzANode *n = get_anode (g->curnode);
-				if (n) {
-					ut64 addr = rz_num_get (NULL, n->title);
-					toggle_bb (core, addr);
-					g->need_reload_nodes = true;
-				}
 			}
 			break;
 		case 's':

--- a/librz/core/canalysis.c
+++ b/librz/core/canalysis.c
@@ -3027,9 +3027,6 @@ static int fcn_print_detail(RzCore *core, RzAnalysisFunction *fcn) {
 	if (fcn->cc || defaultCC) {
 		rz_cons_printf ("afc %s @ 0x%08"PFMT64x"\n", fcn->cc?fcn->cc: defaultCC, fcn->addr);
 	}
-	if (fcn->folded) {
-		rz_cons_printf ("afF @ 0x%08"PFMT64x"\n", fcn->addr);
-	}
 	if (fcn) {
 		/* show variables  and arguments */
 		rz_core_cmdf (core, "afvb* @ 0x%"PFMT64x"\n", fcn->addr);

--- a/librz/core/cconfig.c
+++ b/librz/core/cconfig.c
@@ -3121,7 +3121,6 @@ RZ_API int rz_core_config_init(RzCore *core) {
 	SETBPREF ("asm.sub.varonly", "true", "Substitute the entire variable expression with the local variable name (e.g. [local10h] instead of [ebp+local10h])");
 	SETBPREF ("asm.sub.reg", "false", "Substitute register names with their associated role name (drp~=)");
 	SETBPREF ("asm.sub.rel", "true", "Substitute pc relative expressions in disasm");
-	SETBPREF ("asm.cmt.fold", "false", "Fold comments, toggle with Vz");
 	SETBPREF ("asm.family", "false", "Show family name in disasm");
 	SETBPREF ("asm.symbol", "false", "Show symbol+delta instead of absolute offset");
 	SETBPREF ("asm.analysis", "false", "Analyze code and refs while disassembling (see analysis.strings)");

--- a/librz/core/cmd_analysis.c
+++ b/librz/core/cmd_analysis.c
@@ -324,12 +324,10 @@ static const char *help_msg_af[] = {
 	"afa", "", "analyze function arguments in a call (afal honors dbg.funcarg)",
 	"afb+", " fcnA bbA sz [j] [f] ([t]( [d]))", "add bb to function @ fcnaddr",
 	"afb", "[?] [addr]", "List basic blocks of given function",
-	"afbF", "([0|1])", "Toggle the basic-block 'folded' attribute",
 	"afB", " 16", "set current function as thumb (change asm.bits)",
 	"afC[lc]", " ([addr])@[addr]", "calculate the Cycles (afC) or Cyclomatic Complexity (afCc)",
 	"afc", "[?] type @[addr]", "set calling convention for function",
 	"afd", "[addr]","show function + delta for given offset",
-	"afF", "[1|0|]", "fold/unfold/toggle",
 	"afi", " [addr|fcn.name]", "show function(s) information (verbose afl)",
 	"afj", " [tableaddr] [count]", "analyze function jumptable",
 	"afl", "[?] [ls*] [fcn name]", "list functions (addr, size, bbs, name) (see afll)",
@@ -3946,24 +3944,6 @@ static int cmd_analysis_fcn(RzCore *core, const char *input) {
 		case 'e': // "afbe"
 			analysis_bb_edge (core, input + 3);
 			break;
-		case 'F': // "afbF"
-			{
-			RzAnalysisFunction *fcn = rz_analysis_get_fcn_in (core->analysis, core->offset, RZ_ANALYSIS_FCN_TYPE_NULL);
-			if (fcn) {
-				RzAnalysisBlock *bb = rz_analysis_fcn_bbget_in (core->analysis, fcn, core->offset);
-				if (bb) {
-					if (input[3]) {
-						int n = atoi (input + 3);
-						bb->folded = n;
-					} else {
-						bb->folded = !bb->folded;
-					}
-				} else {
-					rz_warn_if_reached ();
-				}
-			}
-			}
-			break;
 		case 0:
 		case ' ': // "afb "
 		case 'q': // "afbq"
@@ -4182,15 +4162,6 @@ static int cmd_analysis_fcn(RzCore *core, const char *input) {
 		default:
 			eprintf ("Wrong command. Look at af?\n");
 			break;
-		}
-		break;
-	case 'F': // "afF"
-		{
-			int val = input[2] && rz_num_math (core->num, input + 2);
-			RzAnalysisFunction *fcn = rz_analysis_get_fcn_in (core->analysis, core->offset, RZ_ANALYSIS_FCN_TYPE_NULL);
-			if (fcn) {
-				fcn->folded = input[2]? val: !fcn->folded;
-			}
 		}
 		break;
 	case '?': // "af?"

--- a/librz/core/core.c
+++ b/librz/core/core.c
@@ -866,7 +866,7 @@ static const char *rizin_argv[] = {
 	"afb?", "afb", "afb.", "afb+", "afbb", "afbr", "afbi", "afbj", "afbe", "afB", "afbc", "afb=",
 	"afB", "afC", "afCl", "afCc", "afc?", "afc", "afc=", "afcr", "afcrj", "afca", "afcf", "afcfj",
 	"afck", "afcl", "afco", "afcR",
-	"afd", "aff", "afF", "afi",
+	"afd", "aff", "afi",
 	"afl?", "afl", "afl+", "aflc", "aflj", "afll", "afllj", "aflm", "aflq", "aflqj", "afls",
 	"afm", "afM", "afn?", "afna", "afns", "afnsj", "afl=",
 	"afo", "afs", "afS", "aft?", "aft", "afu",

--- a/librz/core/disasm.c
+++ b/librz/core/disasm.c
@@ -100,7 +100,6 @@ typedef struct {
 	bool linesright;
 	int tracespace;
 	int cyclespace;
-	int cmtfold;
 	int show_indent;
 	bool show_dwarf;
 	bool show_size;
@@ -735,7 +734,6 @@ static RDisasmState * ds_init(RzCore *core) {
 	ds->show_stackptr = rz_config_get_i (core->config, "asm.stackptr");
 	ds->show_xrefs = rz_config_get_i (core->config, "asm.xrefs");
 	ds->show_cmtrefs = rz_config_get_i (core->config, "asm.cmt.refs");
-	ds->cmtfold = rz_config_get_i (core->config, "asm.cmt.fold");
 	ds->show_cmtoff = rz_config_get (core->config, "asm.cmt.off");
 	if (!ds->show_cmtoff) {
 		ds->show_cmtoff = "nodup";
@@ -2118,23 +2116,12 @@ static void ds_show_comments_right(RDisasmState *ds) {
 		if (ds->show_color) {
 			rz_cons_strcat (ds->pal_comment);
 		}
-		/* print multiline comment */
-		if (ds->cmtfold) {
-			char *p = strdup (ds->comment);
-			char *q = strchr (p, '\n');
-			if (q) {
-				*q = 0;
-				rz_cons_strcat (p);
-				rz_cons_strcat (" ; [z] unfold");
-			}
-			free (p);
-		} else {
-			ds_pre_xrefs (ds, false);
-			if (ds->show_color) {
-				rz_cons_strcat (ds->color_usrcmt);
-			}
-			ds_comment (ds, false, "%s", ds->comment);
+		ds_pre_xrefs (ds, false);
+		if (ds->show_color) {
+			rz_cons_strcat (ds->color_usrcmt);
 		}
+		ds_comment (ds, false, "%s", ds->comment);
+
 		if (ds->show_color) {
 			ds_print_color_reset (ds);
 		}
@@ -5134,7 +5121,6 @@ static void ds_end_line_highlight(RDisasmState *ds) {
 // int l is for lines
 RZ_API int rz_core_print_disasm(RzPrint *p, RzCore *core, ut64 addr, ut8 *buf, int len, int l, int invbreak, int cbytes, bool json, PJ *pj, RzAnalysisFunction *pdf) {
 	int continueoninvbreak = (len == l) && invbreak;
-	RzAnalysisFunction *of = NULL;
 	RzAnalysisFunction *f = NULL;
 	bool calc_row_offsets = p->calc_row_offsets;
 	int ret, i, inc = 0, skip_bytes_flag = 0, skip_bytes_bb = 0, idx = 0;
@@ -5285,46 +5271,6 @@ toro:
 		}
 		// f = rz_analysis_get_fcn_in (core->analysis, ds->at, RZ_ANALYSIS_FCN_TYPE_NULL);
 		f = ds->fcn = fcnIn (ds, ds->at, RZ_ANALYSIS_FCN_TYPE_NULL);
-		if (f && f->folded && rz_analysis_function_contains (f, ds->at)) {
-			int delta = (ds->at <= f->addr) ? (ds->at - rz_analysis_function_max_addr (f)) : 0;
-			if (of != f) {
-				char cmt[32];
-				get_bits_comment (core, f, cmt, sizeof (cmt));
-				const char *comment = rz_meta_get_string (core->analysis, RZ_META_TYPE_COMMENT, ds->at);
-				if (comment) {
-					ds_pre_xrefs (ds, true);
-					rz_cons_printf ("; %s\n", comment);
-				}
-				rz_cons_printf ("%s%s%s (fcn) %s%s%s\n",
-					COLOR (ds, color_fline), core->cons->vline[CORNER_TL],
-					COLOR (ds, color_fname), f->name, cmt, COLOR_RESET (ds));
-				ds_setup_print_pre (ds, true, false);
-				ds_print_lines_left (ds);
-				ds_print_offset (ds);
-				rz_cons_printf ("(%" PFMT64d " byte folded function)\n", rz_analysis_function_linear_size (f));
-				//rz_cons_printf ("%s%s%s\n", COLOR (ds, color_fline), core->cons->vline[RDWN_CORNER], COLOR_RESET (ds));
-				if (delta < 0) {
-					delta = -delta;
-				}
-				ds->addr += delta + idx;
-				rz_io_read_at (core->io, ds->addr, buf, len);
-				inc = 0; //delta;
-				idx = 0;
-				of = f;
-				rz_analysis_op_fini (&ds->analop);
-				if (len == l) {
-					break;
-				}
-			} else {
-				ds->lines--;
-				ds->addr += 1;
-				rz_io_read_at (core->io, ds->addr, buf, len);
-				inc = 0; //delta;
-				idx = 0;
-				rz_analysis_op_fini (&ds->analop);
-			}
-			continue;
-		}
 		ds_show_comments_right (ds);
 		// TRY adding here
 		char *link_key = sdb_fmt ("link.%08"PFMT64x, ds->addr + idx);
@@ -5454,13 +5400,6 @@ toro:
 				fcn = rz_analysis_get_function_at (core->analysis, ds->at);
 				if (fcn) {
 					bb = rz_analysis_fcn_bbget_in (core->analysis, fcn, ds->at);
-				}
-			}
-			if (bb) {
-				if (bb->folded) {
-					rz_cons_printf ("[+] Folded BB [..0x%08"PFMT64x"]\n", ds->at + bb->size);
-					inc = bb->size;
-					continue;
 				}
 			}
 		}

--- a/librz/core/visual.c
+++ b/librz/core/visual.c
@@ -336,7 +336,6 @@ static const char *help_msg_visual[] = {
 	"wW", "seek cursor to next/prev word",
 	"xX", "show xrefs/refs of current function from/to data/code",
 	"yY", "copy and paste selection",
-	"z", "fold/unfold comments in disassembly",
 	"Z", "shift-tab rotate print modes", // ctoggle zoom mode",
 	"Enter", "follow address of jump/call",
 	NULL
@@ -2290,20 +2289,6 @@ static bool canWrite(RzCore *core, ut64 addr) {
 	return (map && (map->perm & RZ_PERM_W));
 }
 
-static bool toggle_bb(RzCore *core, ut64 addr) {
-	RzAnalysisFunction *fcn = rz_analysis_get_fcn_in (core->analysis, addr, RZ_ANALYSIS_FCN_TYPE_NULL);
-	if (fcn) {
-		RzAnalysisBlock *bb = rz_analysis_fcn_bbget_in (core->analysis, fcn, addr);
-		if (bb) {
-			bb->folded = !bb->folded;
-		} else {
-			rz_warn_if_reached ();
-		}
-		return true;
-	}
-	return false;
-}
-
 RZ_API int rz_core_visual_cmd(RzCore *core, const char *arg) {
 	ut8 och = arg[0];
 	RzAsmOp op;
@@ -3502,39 +3487,6 @@ RZ_API int rz_core_visual_cmd(RzCore *core, const char *arg) {
 			}
 		}
 		break;
-		case 'z':
-		{
-			RzAnalysisFunction *fcn;
-			if (core->print->cur_enabled) {
-				fcn = rz_analysis_get_fcn_in (core->analysis,
-					core->offset + core->print->cur, RZ_ANALYSIS_FCN_TYPE_NULL);
-			} else {
-				fcn = rz_analysis_get_fcn_in (core->analysis,
-					core->offset, RZ_ANALYSIS_FCN_TYPE_NULL);
-			}
-			if (fcn) {
-				fcn->folded = !fcn->folded;
-			} else {
-				rz_config_toggle (core->config, "asm.cmt.fold");
-			}
-		}
-		break;
-		case 'Z': // shift-tab SHIFT-TAB
-			if (och == 27) { // shift-tab
-				if (core->print->cur_enabled && core->printidx == RZ_CORE_VISUAL_MODE_DB) {
-					core->print->cur = 0;
-					core->seltab--;
-					if (core->seltab < 0) {
-						core->seltab = 2;
-					}
-				} else {
-					prevPrintFormat (core);
-				}
-			} else { // "Z"
-				ut64 addr = core->print->cur_enabled? core->offset + core->print->cur: core->offset;
-				toggle_bb (core, addr);
-			}
-			break;
 		case '?':
 			if (visual_help (core) == '?') {
 				rz_core_visual_hud (core);
@@ -4095,43 +4047,29 @@ static void visual_refresh_oneshot(RzCore *core) {
 }
 
 RZ_API void rz_core_visual_disasm_up(RzCore *core, int *cols) {
-	RzAnalysisFunction *f = rz_analysis_get_fcn_in (core->analysis, core->offset, RZ_ANALYSIS_FCN_TYPE_NULL);
-	if (f && f->folded) {
-		*cols = core->offset - f->addr; // + f->size;
-		if (*cols < 1) {
-			*cols = 4;
-		}
-	} else {
-		*cols = rz_core_visual_prevopsz (core, core->offset);
-	}
+	*cols = rz_core_visual_prevopsz (core, core->offset);
 }
 
 RZ_API void rz_core_visual_disasm_down(RzCore *core, RzAsmOp *op, int *cols) {
 	int midflags = rz_config_get_i (core->config, "asm.flags.middle");
 	const bool midbb = rz_config_get_i (core->config, "asm.bb.middle");
-	RzAnalysisFunction *f = NULL;
-	f = rz_analysis_get_fcn_in (core->analysis, core->offset, 0);
 	op->size = 1;
-	if (f && f->folded) {
-		*cols = core->offset - rz_analysis_function_max_addr (f);
-	} else {
-		rz_asm_set_pc (core->rasm, core->offset);
-		*cols = rz_asm_disassemble (core->rasm,
-				op, core->block, 32);
-		if (midflags || midbb) {
-			int skip_bytes_flag = 0, skip_bytes_bb = 0;
-			if (midflags >= RZ_MIDFLAGS_REALIGN) {
-				skip_bytes_flag = rz_core_flag_in_middle (core, core->offset, *cols, &midflags);
-			}
-			if (midbb) {
-				skip_bytes_bb = rz_core_bb_starts_in_middle (core, core->offset, *cols);
-			}
-			if (skip_bytes_flag) {
-				*cols = skip_bytes_flag;
-			}
-			if (skip_bytes_bb && skip_bytes_bb < *cols) {
-				*cols = skip_bytes_bb;
-			}
+	rz_asm_set_pc (core->rasm, core->offset);
+	*cols = rz_asm_disassemble (core->rasm,
+			op, core->block, 32);
+	if (midflags || midbb) {
+		int skip_bytes_flag = 0, skip_bytes_bb = 0;
+		if (midflags >= RZ_MIDFLAGS_REALIGN) {
+			skip_bytes_flag = rz_core_flag_in_middle (core, core->offset, *cols, &midflags);
+		}
+		if (midbb) {
+			skip_bytes_bb = rz_core_bb_starts_in_middle (core, core->offset, *cols);
+		}
+		if (skip_bytes_flag) {
+			*cols = skip_bytes_flag;
+		}
+		if (skip_bytes_bb && skip_bytes_bb < *cols) {
+			*cols = skip_bytes_bb;
 		}
 	}
 	if (*cols < 1) {

--- a/librz/include/rz_analysis.h
+++ b/librz/include/rz_analysis.h
@@ -304,7 +304,6 @@ typedef struct rz_analysis_function_t {
 	st64 stack;  // stack frame size
 	int maxstack;
 	int ninstr;
-	bool folded;
 	bool is_pure;
 	bool is_variadic;
 	bool has_changed; // true if function may have changed since last anaysis TODO: set this attribute where necessary
@@ -915,7 +914,6 @@ typedef struct rz_analysis_bb_t {
 	ut64 jump;
 	ut64 fail;
 	bool traced;
-	bool folded;
 	ut32 colorize;
 	ut8 *fingerprint;
 	RzAnalysisDiff *diff;

--- a/test/db/analysis/x86_32
+++ b/test/db/analysis/x86_32
@@ -3023,46 +3023,6 @@ ret
 EOF
 RUN
 
-NAME=af afF
-FILE=malloc://1024
-CMDS=<<EOF
-e analysis.hasnext=0
-e analysis.eobjmp=false
-e asm.arch=x86
-e asm.bits=32
-wx 6a004889e5c3
-af test
-afF
-pd 1 @ test
-EOF
-EXPECT=<<EOF
-/ (fcn) test
-\           0x00000000 (6 byte folded function)
-EOF
-RUN
-
-NAME=af afF graph
-FILE=malloc://1024
-CMDS=<<EOF
-e analysis.hasnext=0
-e analysis.eobjmp=false
-e asm.arch=x86
-e asm.bits=32
-wx 6a004889e5c3
-af test
-afF
-agf
-EOF
-EXPECT=<<EOF
-[0x00000000]>  # test ();
- --------------------------.
-|  0x0                     |
-| / (fcn) test             |
-| (6 byte folded function) |
-`--------------------------'
-EOF
-RUN
-
 NAME=ahc
 FILE=malloc://16
 CMDS=<<EOF

--- a/test/unit/test_serialize_analysis.c
+++ b/test/unit/test_serialize_analysis.c
@@ -140,7 +140,7 @@ bool test_analysis_switch_op_load() {
 Sdb *blocks_ref_db() {
 	Sdb *db = sdb_new0 ();
 	sdb_set (db, "0x539", "{\"size\":42}", 0);
-	sdb_set (db, "0x4d2", "{\"size\":32,\"jump\":4883,\"fail\":16915,\"traced\":true,\"folded\":true,\"colorize\":16711680,\"fingerprint\":\"AAECAwQFBgcICQoLDA0ODxAREhMUFRYXGBkaGxwdHh8=\",\"diff\":{\"addr\":54123},\"switch_op\":{\"addr\":49232,\"min\":3,\"max\":5,\"def\":7,\"cases\":[]},\"ninstr\":3,\"op_pos\":[4,7],\"stackptr\":43,\"parent_stackptr\":57,\"cmpval\":3735928559,\"cmpreg\":\"rax\"}", 0);
+	sdb_set (db, "0x4d2", "{\"size\":32,\"jump\":4883,\"fail\":16915,\"traced\":true,\"colorize\":16711680,\"fingerprint\":\"AAECAwQFBgcICQoLDA0ODxAREhMUFRYXGBkaGxwdHh8=\",\"diff\":{\"addr\":54123},\"switch_op\":{\"addr\":49232,\"min\":3,\"max\":5,\"def\":7,\"cases\":[]},\"ninstr\":3,\"op_pos\":[4,7],\"stackptr\":43,\"parent_stackptr\":57,\"cmpval\":3735928559,\"cmpreg\":\"rax\"}", 0);
 	return db;
 }
 
@@ -153,7 +153,6 @@ bool test_analysis_block_save() {
 	block->jump = 0x1313;
 	block->fail = 0x4213;
 	block->traced = true;
-	block->folded = true;
 	block->colorize = 0xff0000;
 	block->fingerprint = malloc (block->size);
 	ut8 v;
@@ -212,7 +211,6 @@ bool test_analysis_block_load() {
 	mu_assert_eq (a->jump, UT64_MAX, "jump");
 	mu_assert_eq (a->fail, UT64_MAX, "fail");
 	mu_assert ("traced", !a->traced);
-	mu_assert ("folded", !a->folded);
 	mu_assert_eq (a->colorize, 0, "colorize");
 	mu_assert_null (a->fingerprint, "fingerprint");
 	mu_assert_null (a->diff, "diff");
@@ -228,7 +226,6 @@ bool test_analysis_block_load() {
 	mu_assert_eq (b->jump, 0x1313, "jump");
 	mu_assert_eq (b->fail, 0x4213, "fail");
 	mu_assert ("traced", b->traced);
-	mu_assert ("folded", b->folded);
 	mu_assert_eq (b->colorize, 0xff0000, "colorize");
 	mu_assert_notnull (b->fingerprint, "fingerprint");
 	mu_assert_memeq (b->fingerprint,
@@ -276,7 +273,7 @@ Sdb *functions_ref_db() {
 	Sdb *db = sdb_new0 ();
 	sdb_set (db, "0x4d2", "{\"name\":\"effekt\",\"type\":1,\"stack\":0,\"maxstack\":0,\"ninstr\":0,\"bp_frame\":true,\"pure\":true,\"diff\":{},\"bbs\":[1337]}", 0);
 	sdb_set (db, "0xbeef", "{\"name\":\"eskapist\",\"bits\":32,\"type\":16,\"stack\":0,\"maxstack\":0,\"ninstr\":0,\"bp_frame\":true,\"diff\":{},\"bbs\":[]}", 0);
-	sdb_set (db, "0x539", "{\"name\":\"hirsch\",\"bits\":16,\"type\":0,\"cc\":\"fancycall\",\"stack\":42,\"maxstack\":123,\"ninstr\":13,\"folded\":true,\"bp_frame\":true,\"bp_off\":4,\"fingerprint\":\"AAECAwQFBgcICQoLDA0ODw==\",\"diff\":{\"addr\":4321},\"bbs\":[1337,1234],\"imports\":[\"earth\",\"rise\"],\"labels\":{\"beach\":1400,\"another\":1450,\"year\":1440}}", 0);
+	sdb_set (db, "0x539", "{\"name\":\"hirsch\",\"bits\":16,\"type\":0,\"cc\":\"fancycall\",\"stack\":42,\"maxstack\":123,\"ninstr\":13,\"bp_frame\":true,\"bp_off\":4,\"fingerprint\":\"AAECAwQFBgcICQoLDA0ODw==\",\"diff\":{\"addr\":4321},\"bbs\":[1337,1234],\"imports\":[\"earth\",\"rise\"],\"labels\":{\"beach\":1400,\"another\":1450,\"year\":1440}}", 0);
 	sdb_set (db, "0xdead", "{\"name\":\"agnosie\",\"bits\":32,\"type\":8,\"stack\":0,\"maxstack\":0,\"ninstr\":0,\"bp_frame\":true,\"diff\":{},\"bbs\":[]}", 0);
 	sdb_set (db, "0xc0ffee", "{\"name\":\"lifnej\",\"bits\":32,\"type\":32,\"stack\":0,\"maxstack\":0,\"ninstr\":0,\"bp_frame\":true,\"diff\":{},\"bbs\":[]}", 0);
 	sdb_set (db, "0x1092", "{\"name\":\"hiberno\",\"bits\":32,\"type\":2,\"stack\":0,\"maxstack\":0,\"ninstr\":0,\"diff\":{},\"bbs\":[]}", 0);
@@ -300,7 +297,6 @@ bool test_analysis_function_save() {
 	f->maxstack = 123;
 	f->bp_off = 4;
 	f->ninstr = 13;
-	f->folded = true;
 	f->fingerprint_size = 0x10;
 	f->fingerprint = malloc (f->fingerprint_size);
 	ut8 v;
@@ -376,7 +372,6 @@ bool test_analysis_function_load() {
 	mu_assert_eq (f->stack, 42, "stack");
 	mu_assert_eq (f->maxstack, 123, "maxstack");
 	mu_assert_eq (f->ninstr, 13, "ninstr");
-	mu_assert ("folded", f->folded);
 	mu_assert ("pure", !f->is_pure);
 	mu_assert ("noreturn", !f->is_noreturn);
 	mu_assert ("bp_frame", f->bp_frame);
@@ -405,7 +400,6 @@ bool test_analysis_function_load() {
 	mu_assert_eq (f->stack, 0, "stack");
 	mu_assert_eq (f->maxstack, 0, "maxstack");
 	mu_assert_eq (f->ninstr, 0, "ninstr");
-	mu_assert ("folded", !f->folded);
 	mu_assert ("pure", f->is_pure);
 	mu_assert ("noreturn", !f->is_noreturn);
 	mu_assert ("bp_frame", f->bp_frame);
@@ -425,7 +419,6 @@ bool test_analysis_function_load() {
 	mu_assert_eq (f->stack, 0, "stack");
 	mu_assert_eq (f->maxstack, 0, "maxstack");
 	mu_assert_eq (f->ninstr, 0, "ninstr");
-	mu_assert ("folded", !f->folded);
 	mu_assert ("pure", !f->is_pure);
 	mu_assert ("noreturn", !f->is_noreturn);
 	mu_assert ("bp_frame", !f->bp_frame);
@@ -444,7 +437,6 @@ bool test_analysis_function_load() {
 	mu_assert_eq (f->stack, 0, "stack");
 	mu_assert_eq (f->maxstack, 0, "maxstack");
 	mu_assert_eq (f->ninstr, 0, "ninstr");
-	mu_assert ("folded", !f->folded);
 	mu_assert ("pure", !f->is_pure);
 	mu_assert ("noreturn", f->is_noreturn);
 	mu_assert ("bp_frame", f->bp_frame);

